src/app/store/test-records/effects/test-records.effects.ts

```diff
@@ -151,12 +155,86 @@ export class TestResultsEffects {
     )
   );
 
+  generateContingencyTestTemplatesAndtestResultToUpdate$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(contingencyTestTypeSelected),
+      mergeMap(action =>
+        of(action).pipe(withLatestFrom(this.store.pipe(select(testResultInEdit)), this.testTypeService.selectAllTestTypes$), take(1))
+      ),
+      concatMap(([action, editingTestResult, testTypesTaxonomy]) => {
+        const { testType } = action;
```

Comment:
This is a pretty long `concatMap` and you can omit this line as `testType` is being used only in one place (line 172).

Reply:
It's also being used on line 185 and 186

Reply:
Ah, my bad.